backend/infrahub/core/schema/schema_branch.py

```diff
@@ -2038,25 +2068,34 @@ def add_relationships_to_template(self, node: NodeSchema | GenericSchema) -> Non
                 if relationship.kind not in [RelationshipKind.ATTRIBUTE, RelationshipKind.GENERIC]
                 else relationship.peer
             )
+
+            is_optional = (
```

Comment:
Not related to this change but while reading the code I found the previous version hard to read